# Given that I can't get to the tweets...

All markdown here is a combo rubber ducky/note taking for when I come back to this. 



In [2]:
import pandas as pd

In [5]:
df = pd.read_csv("Data/users.csv")

In [6]:
df

user_id  user_community user_active_status  \
0                       6590             0.0          suspended   
1                      10597             NaN            deleted   
2                      16943             NaN          suspended   
3                      29803             NaN          suspended   
4                      68093             0.0          suspended   
...                      ...             ...                ...   
2559013   899317849243693057             2.0          suspended   
2559014  1296302672077758464             2.0          suspended   
2559015           1403510982             1.0            deleted   
2559016  1331876413926936577             1.0          suspended   
2559017  1269104794511147008             1.0            deleted   

         closeness_centrality_detractor_cluster  \
0                                           NaN   
1                                           NaN   
2                                           NaN   
3                                           NaN   
4                                      0.076194   
...                                         ...   
2559013                                     NaN   
2559014                                     NaN   
2559015                                     NaN   
2559016                                     NaN   
2559017                                     NaN   

         closeness_centrality_promoter_cluster  retweet_count_by_community_0  \
0                                          NaN                             0   
1                                          NaN                             0   
2                                          NaN                             0   
3                                          NaN                             0   
4                                          NaN                            16   
...                                        ...                           ...   
2559013                                    NaN                             0   
2559014                                    NaN                             0   
2559015                                    NaN                             0   
2559016                                    NaN                             0   
2559017                                    NaN                             0   

         quote_count_by_community_0  retweet_count_by_community_1  \
0                                 0                             0   
1                                 0                             0   
2                                 0                             0   
3                                 0                             0   
4                                 0                             0   
...                             ...                           ...   
2559013                           0                             0   
2559014                           0                             0   
2559015                           0                             0   
2559016                           0                             0   
2559017                           0                             0   

         quote_count_by_community_1  retweet_count_by_community_2  \
0                                 0                             0   
1                                 0                             0   
2                                 0                             0   
3                                 0                             0   
4                                 0                             0   
...                             ...                           ...   
2559013                           0                             0   
2559014                           0                             0   
2559015                           0                             0   
2559016                           0                             0   
2559017                           0                   

Okay... so, user community is the label VF2020 used to classify their communities. 

"To detect communities within the graph, we used the Infomap community detection algorithm, which captures the flow of information in directed networks. Using the default parameters, the algorithm produces thousands of clusters. By excluding all clusters that contain fewer than 1% of the nodes we are left with 90% of all nodes which are clustered into five communities."

Communities 1-4 are all 'promoters' of misinformation, while community 2 is the most suspended group, indicating that Twitter at least deems those types to be the most likely to require moderation.

Let's look at just community 2

In [7]:
df_community_2 = df[df.user_community == 2]

In [8]:
df_community_2

user_id  user_community user_active_status  \
5                     117303             2.0          suspended   
7                     754607             2.0          suspended   
10                    817615             2.0          suspended   
13                    945481             2.0            deleted   
14                   1124241             2.0            deleted   
...                      ...             ...                ...   
2558996   802630862240153600             2.0          suspended   
2559007           1587507343             2.0          suspended   
2559010  1270504542501842944             2.0          suspended   
2559013   899317849243693057             2.0          suspended   
2559014  1296302672077758464             2.0          suspended   

         closeness_centrality_detractor_cluster  \
5                                           NaN   
7                                           NaN   
10                                          NaN   
13                                          NaN   
14                                          NaN   
...                                         ...   
2558996                                     NaN   
2559007                                     NaN   
2559010                                     NaN   
2559013                                     NaN   
2559014                                     NaN   

         closeness_centrality_promoter_cluster  retweet_count_by_community_0  \
5                                          NaN                             0   
7                                          NaN                             0   
10                                         NaN                             0   
13                                         NaN                             0   
14                                         NaN                             0   
...                                        ...                           ...   
2558996                                    NaN                             0   
2559007                                    NaN                             0   
2559010                                    NaN                             0   
2559013                                    NaN                             0   
2559014                                    NaN                             0   

         quote_count_by_community_0  retweet_count_by_community_1  \
5                                 0                             0   
7                                 0                             0   
10                                0                             0   
13                                0                             0   
14                                0                             0   
...                             ...                           ...   
2558996                           0                             0   
2559007                           0                             0   
2559010                           0                             0   
2559013                           0                             0   
2559014                           0                             0   

         quote_count_by_community_1  retweet_count_by_community_2  \
5                                 0                             1   
7                                 0                             0   
10                                0                            15   
13                                0                             0   
14                                0                             3   
...                             ...                           ...   
2558996                           0                             0   
2559007                           0                             0   
2559010                           0                             0   
2559013                           0                             0   
2559014                           0                   

Unfortunately, tweets from suspended or deleted accounts are not publically available (the VF2020 folks have a database, but are only sharing with published acedemics.) Let's look at the active status column.

In [9]:
df_community_2.user_active_status.value_counts()

active       266332
suspended     46031
deleted       29821
Name: user_active_status, dtype: int64

In [11]:
df_community_2 = df_community_2[df_community_2.user_active_status== 'active']

In [12]:
df_community_2.user_active_status.value_counts()

active    266332
Name: user_active_status, dtype: int64

In [13]:
df_community_2

user_id  user_community user_active_status  \
19                   1656021             2.0             active   
210                 14154658             2.0             active   
386                 14884533             2.0             active   
721                 16047786             2.0             active   
733                 16083632             2.0             active   
...                      ...             ...                ...   
2488665             78739450             2.0             active   
2488678  1311449369028308992             2.0             active   
2488686  1311450324352282629             2.0             active   
2488716           3828574157             2.0             active   
2488719  1311455302034829312             2.0             active   

         closeness_centrality_detractor_cluster  \
19                                          NaN   
210                                         NaN   
386                                         NaN   
721                                         NaN   
733                                         NaN   
...                                         ...   
2488665                                     NaN   
2488678                                     NaN   
2488686                                     NaN   
2488716                                     NaN   
2488719                                     NaN   

         closeness_centrality_promoter_cluster  retweet_count_by_community_0  \
19                                         NaN                             0   
210                                        NaN                             0   
386                                        NaN                             0   
721                                        NaN                             0   
733                                        NaN                             0   
...                                        ...                           ...   
2488665                                    NaN                             0   
2488678                                    NaN                             0   
2488686                                    NaN                             0   
2488716                                    NaN                             0   
2488719                                    NaN                             0   

         quote_count_by_community_0  retweet_count_by_community_1  \
19                                0                             0   
210                               0                             0   
386                               0                             0   
721                               0                             0   
733                               0                             0   
...                             ...                           ...   
2488665                           0                             0   
2488678                           0                             0   
2488686                           0                             0   
2488716                           0                             0   
2488719                           0                             0   

         quote_count_by_community_1  retweet_count_by_community_2  \
19                                0                             2   
210                               0                             3   
386                               0                             0   
721                               0                             1   
733                               0                             2   
...                             ...                           ...   
2488665                           0                             0   
2488678                           0                             0   
2488686                           0                             0   
2488716                           0                             0   
2488719                           0                   

Okay, that's 266332 active users from community 2.

Okay, philosophical question here, assuming I can scrape these, do I focus on their tweets in general, or on just the tweets in the VF2020 dataset?

In other words, am I trying to identify a type of person (someone spiralling into or flirting with radicalization) or a type of tweet? I think I might not know until I can read some of the specific tweets that vf2020 have highlighted. 

# Okay, what next?

In [15]:
nov_tweet_df = pd.read_csv('Data/tweets-2020-11.csv')

In [16]:
nov_tweet_df

tweet_id  user_community user_active_status  \
0        1322689823212212227             2.0             active   
1        1322689842552197125             0.0             active   
2        1322689865302085638             4.0             active   
3        1322689872243601409             2.0             active   
4        1322689874022027264             0.0             active   
...                      ...             ...                ...   
5067165  1333561416842874881             2.0          suspended   
5067166  1333561419233595392             0.0             active   
5067167  1333561423746752513             0.0             active   
5067168  1333561440754655232             1.0             active   
5067169  1333561446815379456             2.0             active   

         retweet_count_metadata  quote_count_metadata  \
0                             0                     0   
1                             2                     0   
2                             1                     0   
3                             0                     0   
4                            15                     0   
...                         ...                   ...   
5067165                      23                     1   
5067166                       1                     0   
5067167                       1                     0   
5067168                       4                     0   
5067169                       1                     0   

         retweet_count_by_community_0  quote_count_by_community_0  \
0                                   0                           0   
1                                   2                           0   
2                                   0                           0   
3                                   0                           0   
4                                  15                           0   
...                               ...                         ...   
5067165                             0                           0   
5067166                             1                           0   
5067167                             1                           0   
5067168                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_1  quote_count_by_community_1  \
0                                   0                           0   
1                                   0                           0   
2                                   0                           0   
3                                   0                           0   
4                                   0                           0   
...                               ...                         ...   
5067165                             0                           0   
5067166                             0                           0   
5067167                             0                           0   
5067168                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_2  quote_count_by_community_2  \
0                                   0                           0   
1                                   0                           0   
2                                   0                           0   
3                                   0                           0   
4                                   1                           0   
...                               ...                         ...   
5067165                            19                           0   
5067166                             0                           0   
5067167                             0                           0   
5067168                             0                           0   
5067169                             1                           0   

         retweet_count_by_community_3  quot

In [17]:
nov_tweet_df = nov_tweet_df[nov_tweet_df.user_community == 2]

In [18]:
nov_tweet_df

tweet_id  user_community user_active_status  \
0        1322689823212212227             2.0             active   
3        1322689872243601409             2.0             active   
5        1322689885099229186             2.0          suspended   
6        1322689887368347655             2.0          suspended   
7        1322689897682161665             2.0          suspended   
...                      ...             ...                ...   
5067160  1333561267760586752             2.0             active   
5067162  1333561357313003520             2.0             active   
5067164  1333561400749338624             2.0             active   
5067165  1333561416842874881             2.0          suspended   
5067169  1333561446815379456             2.0             active   

         retweet_count_metadata  quote_count_metadata  \
0                             0                     0   
3                             0                     0   
5                             0                     0   
6                             0                     0   
7                             0                     0   
...                         ...                   ...   
5067160                       3                     0   
5067162                       1                     0   
5067164                       1                     0   
5067165                      23                     1   
5067169                       1                     0   

         retweet_count_by_community_0  quote_count_by_community_0  \
0                                   0                           0   
3                                   0                           0   
5                                   0                           0   
6                                   0                           0   
7                                   0                           0   
...                               ...                         ...   
5067160                             0                           0   
5067162                             0                           0   
5067164                             0                           0   
5067165                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_1  quote_count_by_community_1  \
0                                   0                           0   
3                                   0                           0   
5                                   0                           0   
6                                   0                           0   
7                                   0                           0   
...                               ...                         ...   
5067160                             0                           0   
5067162                             0                           0   
5067164                             0                           0   
5067165                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_2  quote_count_by_community_2  \
0                                   0                           0   
3                                   0                           0   
5                                   0                           0   
6                                   0                           0   
7                                   0                           0   
...                               ...                         ...   
5067160                             1                           0   
5067162                             1                           0   
5067164                             1                           0   
5067165                            19                           0   
5067169                             1                           0   

         retweet_count_by_community_3  quot

In [21]:
nov_tweet_df = nov_tweet_df[nov_tweet_df.user_active_status == 'active']

In [22]:
nov_tweet_df

tweet_id  user_community user_active_status  \
0        1322689823212212227             2.0             active   
3        1322689872243601409             2.0             active   
13       1322689940799574017             2.0             active   
21       1322690008478851078             2.0             active   
22       1322690018368987136             2.0             active   
...                      ...             ...                ...   
5067159  1333561186168807426             2.0             active   
5067160  1333561267760586752             2.0             active   
5067162  1333561357313003520             2.0             active   
5067164  1333561400749338624             2.0             active   
5067169  1333561446815379456             2.0             active   

         retweet_count_metadata  quote_count_metadata  \
0                             0                     0   
3                             0                     0   
13                            0                     0   
21                            0                     0   
22                            0                     0   
...                         ...                   ...   
5067159                       3                     0   
5067160                       3                     0   
5067162                       1                     0   
5067164                       1                     0   
5067169                       1                     0   

         retweet_count_by_community_0  quote_count_by_community_0  \
0                                   0                           0   
3                                   0                           0   
13                                  0                           0   
21                                  0                           0   
22                                  0                           0   
...                               ...                         ...   
5067159                             0                           0   
5067160                             0                           0   
5067162                             0                           0   
5067164                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_1  quote_count_by_community_1  \
0                                   0                           0   
3                                   0                           0   
13                                  0                           0   
21                                  0                           0   
22                                  0                           0   
...                               ...                         ...   
5067159                             0                           0   
5067160                             0                           0   
5067162                             0                           0   
5067164                             0                           0   
5067169                             0                           0   

         retweet_count_by_community_2  quote_count_by_community_2  \
0                                   0                           0   
3                                   0                           0   
13                                  0                           0   
21                                  0                           0   
22                                  0                           0   
...                               ...                         ...   
5067159                             1                           0   
5067160                             1                           0   
5067162                             1                           0   
5067164                             1                           0   
5067169                             1                           0   

         retweet_count_by_community_3  quot

In [23]:
nov_tweet_df.retweet_count_metadata.value_counts()

0       1384418
1        178115
2         55917
3         27068
4         16105
         ...   
574           1
571           1
568           1
563           1
1017          1
Name: retweet_count_metadata, Length: 808, dtype: int64

In [ ]:
1,724,986 